### DISCUSSION TITLE
Asking for permission vs forgiveness

### DISCUSSION THREAD

**Discussion by BishalN** [prisma/prisma#24712] [Q&A]
### Which one of this approach is better and performant

```js
// Permission approach
const comment = await db.comment.findUnique({
  where: { id: Number(commentId) },
});
if (!comment) {
  return c.json({ message: "Comment not found" }, 404);
}
if (comment.userId !== currentUser.id) {
  return c.json({ message: "Unauthorized" }, 401);
}
await db.comment.delete({
  where: { id: Number(commentId) },
});

// Forgiveness approach
// Here no need to check if it exists or not, as well as authorization is done in single step
// Wrap this up in try catch block
  await db.comment.delete({
    where: { id: Number(commentId), userId: currentUser.id },
  });
```

PS: There are no special index everything is default

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @BishalN 👋 

Since there are no special indices and everything is default, the performance difference may not be significant for small-scale applications. However, for large-scale applications, reducing the number of database queries can lead to better performance and scalability. In the permission approach, you make two separate calls to the database:

- One to find the comment by ID.
- Another to delete the comment (if it exists and the user is authorized). 

The forgiveness approach combines these checks into a single delete operation with a specific WHERE clause. This reduces database calls, improving performance.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
